autorag/data/parse/table_hybrid_parse.py

```diff
@@ -0,0 +1,100 @@
+import os
+import tempfile
+from glob import glob
+from typing import List, Tuple, Dict
+
+from PyPDF2 import PdfFileReader, PdfFileWriter
+import pdfplumber
+
+from autorag.support import get_support_modules
+from autorag.data.parse.base import parser_node
+
+
+@parser_node
+def table_hybrid_parse(
+	data_path_list: List[str],
+	text_parse_module: str,
+	text_params: Dict,
+	table_parse_module: str,
+	table_params: Dict,
+) -> Tuple[List[str], List[str]]:
+	# make save folder directory
+	with tempfile.TemporaryDirectory() as save_dir:
+		text_dir = os.path.join(save_dir, "text")
+		table_dir = os.path.join(save_dir, "table")
+
+		os.makedirs(text_dir, exist_ok=True)
+		os.makedirs(table_dir, exist_ok=True)
+
+		# Split PDF file into pages and Save PDFs with and without tables
+		for data_path in data_path_list:
+			save_page_by_table(data_path, text_dir, table_dir)
+
+		# Extract text pages
+		table_results, table_file_names = get_each_module_result(
+			table_parse_module, table_params, os.path.join(table_dir, "*")
+		)
+
+		# Extract table pages
+		text_results, text_file_names = get_each_module_result(
+			text_parse_module, text_params, os.path.join(text_dir, "*")
+		)
+
+		# Merge parsing results of PDFs with and without tables
+		texts = table_results + text_results
+		file_names = table_file_names + text_file_names
```

Comment:
Even though the file_names are in the temp directory, they go through `file_name = os.path.basename(data_path)`, so I think it's okay because it only stores the .pdf filename.

For example, if I have a 3-page `Newjeans_book.pdf`, 
file_names will only contain ['Newjeans_book_page_1.pdf`, 'Newjeans_book_page_2.pdf`, 'Newjeans_book_page_3.pdf`] !

Reply:
I think it is tricky to trace back the original pdf when the hybrid parser returns that.

1. The parser must have similar input and output schema like another parsers. Agree?
2. Let's imagine we want to make a GUI visualizer that want to look parsing result easily. We have to load each parsing result, and also the origin file. 
If the file name looks like this 'havertz_0_1.pdf' then the hybrid parser filename result looks like this. 'havertz_0_1_1.pdf' for first page.
Then how can you know the pdf original name is 'havertz_0_1.pdf' or 'havertz_0_1_1.pdf'??
Plus, even worse, what if there is another pdf that has different meaning that has name 'havertz_0_1_1.pdf' alerady?
3. So, in my opinion, the output information of where is the file comes from **must be** traceable. 
4. The easiest option will be returning each page number, and the whole file will be -1.